File: golang/3_longest_substring_without_repeating_characters.go
package main

import (
	"fmt"
	"strings"
)

func lengthOfLongestSubstring(s string) int {
	var res int = 0
	var st string = ""
	for _, c := range s {
		if strings.Contains(st, string(c)) {
			ind := strings.Index(st, string(c))
			st = st[ind+1:]
		}
		st += string(c)
		if len(st) > res {
			res = len(st)
		}
	}
	return res
}

func main() {
	var inp string
	fmt.Scanln(&inp)
	fmt.Println(lengthOfLongestSubstring(inp))
}

File: python/3_longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        substr = ''
        for ch in s:
            if ch not in substr:
                substr += ch
            else:
                ind = substr.find(ch)
                substr = substr[ind + 1:]
                substr += ch
            res = max(res, len(substr))
        return res

sol = Solution()
inp = input()
print(sol.lengthOfLongestSubstring(inp))